server/app/routes/admin.products.tsx

```diff
@@ -194,46 +264,66 @@ export default function Index() {
   );
 }
 
-function getProductBoxes(
-  products: SerializeFrom<ProductData>[],
-  submit: SubmitFunction
-) {
-  return products.map((product) => {
-    return (
-      <Card key={product.id} className="m-5 grow basis-72">
-        <Title>{product.name}</Title>
-        <div className="my-3">
-          <div>
-            <b>ID</b>: {product.id}
-          </div>
-          <div>
-            <b>Name</b>: {product.name}
-          </div>
-          <div>
-            <b>Reviews</b>: {product.reviewCount}
-          </div>
-          <div>
-            <b>Reports</b>: {product.reportCount}
-          </div>
-          <div>
-            <b>Created on</b>: {product.createdAt}
-          </div>
-          <div>
-            <b>Updated on</b>: {product.updatedAt}
-          </div>
+interface AdminProductCardProps {
+  product: SerializeFrom<ProductData>;
+  submit: SubmitFunction;
+  action: string;
+  setAction: (action: string) => void;
+  isSubmitting: boolean;
+}
+
+function AdminProductCard({
+  product,
+  submit,
+  action,
+  setAction,
+  isSubmitting,
+}: AdminProductCardProps) {
+  const navigation = useNavigation();
+  const isNavProdInfo =
+    navigation.state === "loading" &&
+    navigation.location.pathname === `/admin/product/${product.id}`;
+  return (
+    <Card className="m-5 grow basis-72">
+      <Title>{product.name}</Title>
+      <div className="my-3">
+        <div>
+          <b>ID</b>: {product.id}
+        </div>
+        <div>
+          <b>Name</b>: {product.name}
+        </div>
+        <div>
+          <b>Reviews</b>: {product.reviewCount}
+        </div>
+        <div>
+          <b>Reports</b>: {product.reportCount}
+        </div>
+        <div>
+          <b>Created on</b>: {product.createdAt}
         </div>
+        <div>
+          <b>Updated on</b>: {product.updatedAt}
+        </div>
+      </div>
 
-        <Link to={`/admin/product/${product.id}`}>
-          <Button className="mt-4 block" size="sm">
-            View Product Information
-          </Button>
+      <div className="flex flex-col gap-2">
+        <Link
+          to={`/admin/product/${product.id}`}
+          className={`${buttonVariants({ size: "sm" })} ${
+            isNavProdInfo ? "disabled pointer-events-none opacity-50" : ""
+          }`}
+        >
+          <InlineLoadingSpinner show={isNavProdInfo} />
+          View Product Information
         </Link>
 
         <Button
           type="submit"
-          className="mt-4 block"
           size="sm"
+          disabled={isSubmitting}
           onClick={() => {
+            setAction(`clear-${product.id}`);
```

Comment:
The loading circle for this action (Clear Reports) is shown even if the user presses Cancel at the confirmation prompt, at which point it will stay visible / not get cleared.

Reply:
Resolved by b9bd9d4e